src/stopping_solve.jl

```diff
@@ -0,0 +1,97 @@
+struct NNStopping{C,O,S,E} <: NeuralNetDiffEqAlgorithm
+    chain::C
+    opt::O
+    sdealg::S
+    ensemblealg::E
+end
+NNStopping(chain  ; opt=Flux.ADAM(0.1) , sdealg = EM() , ensemblealg = EnsembleThreads()) = NNStopping(chain , opt , sdealg , ensemblealg)
+
+function DiffEqBase.solve(
+    prob::OptimalStoppingProblem,
+    alg::NeuralNetDiffEqAlgorithm;
+    abstol = 1f-6,
+    verbose = false,
+    maxiters = 300,
+    trajectories = 1000,
+    save_everystep = false,
+    dt,
+    kwargs...
+    )
+
+    tspan = prob.tspan
+    sigma = prob.sigma
+    μ = prob.mu
+    g = prob.g
+    u0 = prob.u0
+    ts = tspan[1]:dt:tspan[2]
+    N = size(ts)[1]
+    T = tspan[2]
+
+    m  = alg.chain
+    opt    = alg.opt
+    sdealg = alg.sdealg
+    ensemblealg = alg.ensemblealg
+    
+    prob = SDEProblem(μ,sigma,u0,tspan)
+    ensembleprob = EnsembleProblem(prob)
+    sim = solve(ensembleprob, sdealg, ensemblealg, dt=dt,trajectories=trajectories,adaptive=false)
+    payoff = []
+    times = []
+    iter = 0
+    sump = 0
+    for u in sim.u
+        X = u.u
+        a = []
+        function Un(n , X )
+            if size(a)[1] >= n
+                return a[n]
+            else
+                if(n == 1)
+                      ans =  first(m(X[1])[1])
+                      a = [ans]
+                      return ans
+                  else
+                      ans =  max(first(m(X[n])[n]) , n + 1 - N)*(1 - sum(Un(i , X ) for i in 1:n-1))
```

Comment:
use a callback for early termination. That should be a lot more efficient: https://docs.sciml.ai/latest/features/callback_functions/#Example-2:-Terminating-an-Integration-1